lib/fhir_models/validation/element_validator/terminology_validator.rb

```diff
@@ -0,0 +1,102 @@
+module FHIR
+  module Validation
+    # Validate Terminology in coded elements
+    class TerminologyValidator
```

Comment:
`TerminologyValidator` needs to maintain state in the way of validators for each code/valueset.  The other validators, like `CardinalityValidator`, are modules because they do not need to maintain any state and are treated as deterministic functions.

The other nice thing is that it helps prevent you from registering `CardinalityValidator` multiple times in a single instance of a `StructureValidator`.